micro/exts.go

```diff
@@ -30,7 +30,12 @@ func exts(x *ast.Variable, v *ast.SExpr, s Substitutions) (Substitutions, bool)
 	if occurs(x, v, s) {
 		return nil, false
 	}
-	return append([]*Substitution{&Substitution{Var: x.Name, Value: v}}, s...), true
+    m := map[string]*ast.SExpr{}
```

Comment:
```
m := deriveClone(s)
m[x.Name] = v
```

Reply:
this panics when s is nil;
can add an explicit assignment to empty map when s is nil but thats almost the same amount of boilerplate.
Do you still prefer using derive in that case?
Looks like:

```
m := deriveClone(s)
if m == nil {
  m = map[string]*ast.SExpr{}
}
m[x.Name] = v
```

Reply:
I think that is a nice bug to log against goderive :)
I want to say until then stick with your way, but deriveClone might still show intent better.
So lets say, either way I am happy.

Reply:
Looking at the deriveClone code, this is actually intentional. 
Cloning nil should result in nil :)
It is however a discrepancy with the standard usecase that clone is supposed to replace here:
you get the empty map when source is nil for free when you write out the loop.
Don't think it's a bug to be reported

Reply:
It makes sense for deriveCopy not to support this, but intuitively I still feel that deriveClone should.